### DISCUSSION TITLE
mjai-reviewer是否可能给Bot提供可用的API？

### DISCUSSION THREAD

**Discussion by KimigaiiWuyi** [Equim-chan/mjai-reviewer#137] [Q&A]
作者你好! 
我有一些关于本项目的疑问，希望能够得到作者的帮助，不胜感激！

我希望为我的开源项目 [MajsoulUID](https://github.com/KimigaiiWuyi/MajsoulUID) 添加一个功能

> 该项目是适用于大部分IM平台的Bot插件（QQ、微信、tg、dc等）

该功能可以通过玩家绑定的UID，向 [牌谱屋](https://amae-koromo.sapk.ch/) 请求玩家近期数据，并同时获取牌谱URL，通过牌谱URL请求mjai-reviewer获取AI分析的json文件，并本地绘制AI分析结果，在IM平台上返回。

- 目前遇到的问题如下：
`mjai.ekyu.moe`网站的可用API，添加了CF的行为验证，所以无法直接请求

我预想中的解决方法有几个，
1. 第一个是请求作者是否可以添加一个需要Token的bot专属API，由搭建项目者自行向作者申请token，以便插件无验证码请求API（或者是以其他可能的方式请求API，比如无验证码但限制QPS (*例如一分钟一次*) 等等）
2. 第二个是自己构建牌谱分析的后端，自行搭建以供插件使用，不过当我检视Mortal的文档的时候，似乎没有发现作者提供的预训练模型（如果自己训练的话，耗时会有点久），以及想自行训练的时候，发现文档大部分条目没有提供（灰色）

如果作者有时间、以及方便的话，能否指导一下两种方法的可能性？
感激不尽
 @Equim-chan

**Comment by herjer** [comment]
这里是牌谱屋的EDWARDH，预训练模型的不公开这点之前已经讨论过，由于预训练模型显然容易变成各平台的机器人核心，所以决定传输计算完的结果给使用者，这点在目前的 环境下大概不太可能改变。
BOT的专属 API 则在合理数量级下个人是赞成的，但也考虑到这可能会变成大量爬谱的漏洞之一，也许可以讨论执行与规范的方式。

**Reply by KimigaiiWuyi** [reply]
对于不参与运营的小规模自建bot来说，个人认为QPS甚至都不需要到1， 大约能支持1分钟一次调用就很好了
而且因为是申请token制的API，其实对于作者本人来说，应该可以定位到更多的异常访问（同个IP大量请求，同个Token大量请求），可能是比较合理的方式
